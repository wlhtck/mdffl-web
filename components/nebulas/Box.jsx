import styled from 'styled-components';
import { white } from './colors';

// TODO: Implement stories with a better component when one exists

export default styled.div`
  display: inline-block;
  margin: 20px;
  padding: 32px;
  background-color: ${({ color }) => color || white};
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
`;
