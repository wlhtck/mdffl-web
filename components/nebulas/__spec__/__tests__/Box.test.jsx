import React from 'react';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import Box from '../../Box';

describe('components/nebulas/box', () => {
  it('renders a <Box />', () => {
    expect(shallow(<Box />)).toMatchSnapshot();
  });
  it('renders a black <Box />', () => {
    expect(shallow(<Box color="#000" />)).toMatchSnapshot();
  });
});
