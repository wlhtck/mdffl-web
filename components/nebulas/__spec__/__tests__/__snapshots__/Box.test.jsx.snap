// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/nebulas/box renders a <Box /> 1`] = `
.c0 {
  display: inline-block;
  margin: 20px;
  padding: 32px;
  background-color: #FFFFFF;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
}

<div
  className="c0"
/>
`;

exports[`components/nebulas/box renders a black <Box /> 1`] = `
.c0 {
  display: inline-block;
  margin: 20px;
  padding: 32px;
  background-color: #000;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
}

<div
  className="c0"
  color="#000"
/>
`;
